core/src/main/java/feast/core/model/FeatureSet.java

```diff
@@ -192,9 +258,11 @@ public void addFeature(Feature feature) {
 
     List<FeatureSpec> featureSpecs = new ArrayList<>();
     for (Feature featureField : features) {
-      FeatureSpec.Builder featureSpecBuilder = FeatureSpec.newBuilder();
-      setFeatureSpecFields(featureSpecBuilder, featureField);
-      featureSpecs.add(featureSpecBuilder.build());
+      if (!featureField.isArchived()) {
```

Comment:
Sure, but is it too much information for the user?

Reply:
I mean if you do something like

`my_feature_set.list_features()` it shouldn't print archived features, but if you do something like `my_feature_set.list_features(show_archived=True)` it should.

We dont have to implement the client side functionality, but it might make sense to still add the archived flag to the message so that its simply a matter of exposing it.

Reply:
I think exposing archived features isn't that trivial, and should be saved for a subsequent PR. I've updated the exception thrown so that users know when they're applying a feature that was previously archived.